src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs

```diff
@@ -87,27 +87,20 @@ protected override async Task HandleWebResourceRequest(CoreWebView2WebResourceRe
 				var uri = new Uri(requestUri);
 				if (new Uri(AppOrigin).IsBaseOf(uri))
 				{
+					var hasRequestBeenHandled = false;
 					var relativePath = new Uri(AppOrigin).MakeRelativeUri(uri).ToString();
 
-					// If the path does not end in a file extension (or is empty), it's most likely referring to a page,
-					// in which case we should allow falling back on the host page.
-					if (allowFallbackOnHostPage && !Path.HasExtension(relativePath))
+					if (Path.HasExtension(relativePath))
 					{
-						relativePath = _hostPageRelativePath;
+						// This looks like a file, so we'll attempt to fetch content using the existing relative path.
+						hasRequestBeenHandled = await TryHandleWinUIStorageWebResourceRequestAsync(eventArgs, relativePath);
 					}
-					relativePath = Path.Combine(_contentRootDir, relativePath.Replace('/', '\\'));
 
-					var winUIItem = await Package.Current.InstalledLocation.TryGetItemAsync(relativePath);
-					if (winUIItem != null)
+					if (!hasRequestBeenHandled && allowFallbackOnHostPage)
 					{
-						statusCode = 200;
-						statusMessage = "OK";
-						var contentType = StaticContentProvider.GetResponseContentTypeOrDefault(relativePath);
-						headers = StaticContentProvider.GetResponseHeaders(contentType);
-						var headerString = GetHeaderString(headers);
-						var stream = await Package.Current.InstalledLocation.OpenStreamForReadAsync(relativePath);
-
-						eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(stream.AsRandomAccessStream(), statusCode, statusMessage, headerString);
+						// Either no content was found at the path, or we didn't attempt to load content because of the path format.
+						// Since we're allowed to fall back on the host page, try to do so.
+						hasRequestBeenHandled = await TryHandleWinUIStorageWebResourceRequestAsync(eventArgs, _hostPageRelativePath);
```

Comment:
It's a bit quirky that we're assigning a value to this flag which is never used.